tcptracer-bpf.h

```diff
@@ -102,6 +107,7 @@ struct perf_event
 	__u16 event_type;
 	__u64 timestamp;
 	union event_payload payload;
+	union connections connection;
```

Comment:
@LouisParkin It looks nicer. Maybe we could also have a `connection_type` here. Then we wouldn't need to have 2 event types for each application protocol. What do you think?

Reply:
The effect on the golang code is the same.  There has to be an `if` or a `switch-case` to decided how we extract the information.  If it is a perf event field indicating network protocol separate from application protocol, I agree, in principle it is the cleaner option.